Modules/+Experiments/@Saturation/run.m

```diff
@@ -1,39 +1,77 @@
-function run(obj,statusH,managers,ax)
-statusWin = statusH.Parent.Parent;
-button = findall(statusWin,'tag','AbortButton');
-newButton = add_button(button,'Stop Acquire');
-newButton.Callback = @obj.stop_acquire;
-set(statusH,'string','Starting data acquisition...');
-drawnow;
+function run( obj,status,managers,ax )
+    % Main run method (callback for CC run button)
+    obj.abort_request = false;
+    status.String = 'Experiment started';
+    ctr = Drivers.Counter.instance(obj.APD_line, obj.APD_sync_line); % Instantiate APD counter
+    drawnow;
 
-panel = ax.Parent;
-delete(ax)
-ax(1) = subplot(1,2,1,'parent',panel);
-ax(2) = subplot(1,2,2,'parent',panel);
+    % Edit this to include meta data for this experimental run (saved in obj.GetData)
+    obj.meta.prefs = obj.prefs2struct;
+    obj.meta.position = managers.Stages.position; % Save current stage position (x,y,z);
+    obj.meta.angles = obj.angle_list; %Angles corresponding to each spectrum
 
-% Flip the acquire boolean to true, set PM wavelength, and initialize data structures
-obj.acquire = true;
-obj.PM100.set_wavelength(obj.wavelength);
-obj.pm_data = [];
-obj.apd_data = [];
+    % Check that rot is not empty and valid
+    assert(~isempty(obj.rot) && isvalid(obj.rot),'Motor SN must be a valid number; motor handle may have been deleted. Check motor serial number/APT Config')
 
-% Continually collect and plot PM and APD data until the user hits "Stop acquisition", which flips
-% obj.acquire to false
-while obj.acquire
-    obj.pm_data = [obj.pm_data, obj.PM100.get_power('MW')];
-    obj.apd_data = [obj.apd_data, obj.counter.singleShot(obj.counter.dwell, obj.nsamples)];
-    [~, sort_index] = sort(obj.pm_data);
-    plot(ax(1),obj.pm_data(sort_index),obj.apd_data(sort_index))
-    xlabel(ax(1),'Input Power (mW)')
-    ylabel(ax(1), 'Output Counts')
-    plot(ax(2),obj.pm_data)
-    ylabel(ax(2),'Input Power (mW)')
-    xlabel(ax(2), 'Collection Bins')
+    % Instantiate data object
+    Nangles = length(obj.angle_list);
+    obj.data.intensity = nan(1,Nangles);
+    
+    % Setup graphics
+    y = NaN(1,Nangles);
+    hold(ax,'on');
+    plotH(1) = plot(obj.angle_list, y,'color', 'k','parent',ax);
+    ylabel(ax,'Counts (cps)');
+    xlabel(ax,['Angle (' char(176) ')']);
+    yyaxis(ax, 'left');
+
+    try
+        % Home rotation mount
+
+        if ~obj.rot.Homed
+            status.String = 'Homing motor'; drawnow;
+            
+            obj.rot.home();
+            pause4Move(obj, obj.motor_home_time);
+        end
+        
+        % Sweep through polarisation and get spectra
+        for i = 1:Nangles
+            theta = obj.angle_list(i);
+            if ~isempty(obj.rot)
+                status.String = sprintf( 'Navigating to %g (%i/%i)', theta, ...
+                    i, Nangles); drawnow;
+                obj.rot.move(theta);
+            else
+                pause(5)
+            end
```

Comment:
Assert above would prevent the `else` from ever happening. This seems to be the only thing that might give a user access to manually do it. But so many other things would break as they currently stand. I recommend just removing manual mode for now.

The easiest way to fix this would be to actually create a manual motor. Then, never change the code, but simply drop in a different motor that calls `questdlg`.